manifests/helm-profiles/compatibility-version-1.24.yaml

```diff
@@ -0,0 +1,4 @@
+pilot:
+  env:
+    # 1.24 behavioral changes
+    ENABLE_INBOUND_RETRY_POLICY: "false"
```

Comment:
I think this is a bit backwards. So this change will land in 1.24.

We need to put this flag in all compatibility-versions *before 1.24*. So then you can install `istio-1.24 install --set compatVersion=1.22` and get the old behavior (no retry)

Reply:
ack. will change. So these 1.24 files need to be renamed to 1.23? I guess the tooling would pick files based on that convention?

Reply:
And copy the same to the 1.21 and 1.22

Reply:
But yeah tooling picks based on the file name convention

Reply:
done. PTAL